src/Hazelcast.Net.Examples/SoakTests/Soak1.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Hazelcast.DistributedObjects;
+using Hazelcast.Examples.Models;
+using Hazelcast.Query;
+using Microsoft.Extensions.Logging;
+
+namespace Hazelcast.Examples.SoakTests
+{
+    // ./hz.ps1 run-example ~Soak1 --%
+    // ./hx.exe ~Soak1
+    //   --hazelcast:examples:soak1:ThreadCount=1             (valid integer)
+    //   --hazelcast:examples:soak1:Duration="HH:MM:SS"       (valid timespan)
+    //   --hazelcast:networking:addresses:0=127.0.0.1:5701
+    //   --hazelcast:networking:addresses:1=127.0.0.1:5702
+
+    // ReSharper disable once UnusedMember.Global
+    public class Soak1
+    {
+        private static int[] _reports;
+
+        private class SoakOptions
+        {
+            public int ThreadCount { get; set; } = 2;
+
+            public string Duration { get; set; } = null;
+
+            public int EntryCount { get; set; } = 10_000;
+        }
+
+        /// <summary>
+        /// Runs the soak test.
+        /// </summary>
+        /// <param name="args"></param>
+        /// <returns>A task that will complete when the soak test has run.</returns>
+        public static async Task Main(string[] args)
+        {
+            // creates the example options
+            var soakOptions = new SoakOptions();
+
+            var options = new HazelcastOptionsBuilder()
+                .With(args)
+                .WithConsoleLogger(LogLevel.Information)
+                .Bind("hazelcast:examples:soak1", soakOptions)
+                .Build();
+
+            var logger = options.LoggerFactory.Service.CreateLogger<Soak1>();
+
+            // say hello
+            logger.LogInformation("Hazelcast .NET soak tests");
+
+            // get duration
+            var duration = TimeSpan.FromMinutes(1); // default to 1 minute
+            if (soakOptions.Duration.EndsWith("h"))
```

Comment:
`soakOptions.Duration` is `null` by default so we get `NullReferenceException` if run without arguments.

Reply:
fixing that one, `soakOptions.Duration` will be `1m` (ie 1 minute) by default